### DISCUSSION TITLE
looking for bdf of a u8g2 font

### DISCUSSION THREAD

**Discussion by ZombieNoSleep** [olikraus/u8g2#2395] [Q&A] [answered]
I am trying to trim down the data size of a font included in u8g2 using bdfconv but I looked in /tools/font/bdf and cannot find it in there.  So, I have 2 questions for anyone that can help please:

1 - can someone point me to the .bdf format of the u8g2_font_6x12_t_symbols 
2 - is there any way to convert the u8g2 .c formatted fonts back to .bdf for editing in Fony, etc?  

Thanks!

**Comment by olikraus** [comment]
The 6x12_t_symbols are a subset of the following font:
https://github.com/olikraus/u8g2/blob/master/tools/font/bdf/6x12.bdf

The precise list of unicode glyphs are listed here:
https://github.com/olikraus/u8g2/blob/ec2559e8466dead8b13dd92cc17ac10e63dab6ac/tools/font/build/build.c#L281

**Reply by ZombieNoSleep** [reply]
just downloaded that bdf and opened in Fony (import, bdf), do not see any of the symbols, am i doing something wrong? 
![fony6x12](https://github.com/olikraus/u8g2/assets/162681799/ddf6f78f-e780-4677-ae42-340e0672bd2b)

**Reply by olikraus** [reply]
The font might be too big for Fony.
Suitable tools might be "gbdfed" (https://github.com/andrewshadura/gbdfed) or fontforge (https://learn.adafruit.com/custom-fonts-for-pyportal-circuitpython-display/conversion)

**Comment by ZombieNoSleep** [comment]
ah yes thanks, looks like Fony only displaying 0-255 and won't let me change above 255, fontforge worked to open it fully....next i tryed bdfconv to reduce 6x12.bdf down to a handful of symbols for testing this process 

bdfconv  -v -f 1 -m "9632-9633,9650-9651,9658-9661,9668-9669" 6x12.bdf -n testfont -o testfont.c

and got the c file:  

const uint8_t testfont[126] U8G2_FONT_SECTION("testfont") = 
  "\12\0\3\2\3\3\1\2\4\6\7\0\0\7\376\12\376\0\0\0\0\0\14A\12}=K\246\15Cf"
  "\13\0\0%\240\7m\35\377!%\241\10m\35\67\333\60%\262\12}]a\66-\303\7%\263\14}"
  "]a\226DIM\33\6%\272\13\356\35\342\20\15\207H\4%\273\13\356\35b\42e\213$\2%\274"
  "\12}\35\277L[\30\1%\304\12\356\235\322p\210\206P%\305\12\356\235\222\262EJ(\0";

put in my code and tried u8g2.setFont(testfont); u8g2.drawGlyph(16,16,9651); for example and not working (display black, doesnt show anything), it works and displays correct character if i change font back to (u8g2_font_6x12_t_symbols)

**Reply by olikraus** [reply]
hmm it should work like this. What is the output of bdfconv?

**Comment by ZombieNoSleep** [comment]
![bdfoutput](https://github.com/olikraus/u8g2/assets/162681799/79680aac-32a7-4f30-9d6d-7e6fff5817e8)

**Reply by ZombieNoSleep** [reply]
what is weird is i can add like capital A, 65 to it in addition to the rest, generate, and can draw.Glyph(x,y,65); just fine but it wont draw the others just black

**Reply by ZombieNoSleep** [reply, marked as answer]
FIXED it!  I was using the wrong bdfconv exe file :) works great now thanks
